docs/src/pages/components/view/demo.tsx

```diff
@@ -26,18 +27,26 @@ const propsToCode = (props: ViewProps) => {
   );
 };
 
+const defaultViewProps = {
+  as: 'div',
+  ariaLabel: 'View example',
+  width: '20rem',
+  height: '4rem',
+  color: 'var(--amplify-colors-blue-60)',
+  backgroundColor: 'var(--amplify-colors-white)',
+  boxShadow: '3px 3px 5px 6px var(--amplify-colors-neutral-60)',
+  padding: '1rem',
+  border: '1px solid var(--amplify-colors-black)',
+  borderRadius: '6px',
+};
+
 export const ViewDemo = () => {
-  const props = useViewProps({
-    as: 'div',
-    ariaLabel: 'View example',
-    width: '20rem',
-    height: '4rem',
-    color: 'var(--amplify-colors-blue-60)',
-    backgroundColor: 'var(--amplify-colors-white)',
-    boxShadow: '3px 3px 5px 6px var(--amplify-colors-neutral-60)',
-    padding: '1rem',
-    border: '1px solid var(--amplify-colors-black)',
-    borderRadius: '6px',
+  const props = useViewProps(
```

Comment:
I think all of our `use[Component]Props` hooks need to be updated to internally use `React.useMemo`, otherwise they will always return a new object literal, which will always run the `useEffect` below even if `props` is added to the dependency array.

**Simplified example**
```jsx
export const useViewProps: UseViewProps = (initialValues) => {
  const [ariaLabel, setAriaLabel] = React.useState<ViewProps['ariaLabel']>(
    initialValues.ariaLabel
  );

  return React.useMemo(() => ({
    ariaLabel,
    setAriaLabel,
  }), [ariaLabel, setAriaLabel);
};
```